### PR TITLE
Handle open and closed requests

### LINKED ISSUE #10: Handle 'opened' and 'closed' requests
## Description

Handle 'opened' as a Create and 'closed' as a Destroy of Pull Requests, but do not delete them from the database, just set the status to 'closed'

## Things to do

* Make a refactor of `find_or_create`
* Catch `opened` and `closed` actions

### PR DESCRIPTION
**Pull request by hdamico**
## Background

Resolves #10

### CODE REVIEW

File: app/services/github_handler.rb [hunk 1]
```diff
@@ -35,50 +39,59 @@ def handle
...
+      draft: pr_data.draft,
+      title: pr_data.title,
+      body: pr_data.body,
+      closed_at: pr_data.closed_at,
+      merged_at: pr_data.merged_at,
+      merged: pr_data.merged,
+      github_id: pr_data.id
+    )
+  rescue ActiveRecord::RecordNotUnique => e
+    errors.add(:base, 'This pull request already exists!')
```
Comment (santib) [inline app/services/github_handler.rb]: what is this error for? the request is triggered by a webhook

File: spec/services/github_handler_spec.rb [hunk 2]
```diff
@@ -48,23 +49,35 @@
...
+    describe 'pull request close' do
+      let!(:pull_request) { create :pull_request, github_id: 1000 }
+
+      before do
+        github_handler.closed
+        pull_request.reload
+      end
+
+      it 'sets status closed' do
+        expect(pull_request.closed?).to be true
```
Comment (santib) [inline spec/services/github_handler_spec.rb]: `expect { github_handler.closed }.to change { pull_request.reload.closed? }.from(false).to(true)`

File: spec/services/github_handler_spec.rb [hunk 3]
```diff
@@ -48,23 +49,35 @@
...
+          github_handler.create_or_find_user(payload.requested_reviewer)
         }.to change(User, :count).by(0)
       end
     end
 
     describe 'pull request not existing' do
       it 'adds pr to the DB' do
         expect {
-          github_handler.create_or_find_pr(payload['pull_request'])
+          github_handler.opened
```
Comment (santib) [inline spec/services/github_handler_spec.rb]: I'd reorganize the test file like

`describe '#opened' do`
`describe '#closed' do`
`describe '#review_requested' do`
`describe '#review_removal' do`

http://www.betterspecs.org/#describe

File: app/services/github_handler.rb (hunk 1 above)
Comment (hdamico) [reply app/services/github_handler.rb]: Do you mean to delete the rescue? I made this for future logs, but I guess it's ok to have the exception on the logs

Comment (santib) [reply app/services/github_handler.rb]: where is `errors` defined? if you wanted to log something maybe you meant `Rails.logger.error('This pull request already exists!')` ?

Comment (hdamico) [reply app/services/github_handler.rb]: Yeah! I mean to do that

File: app/services/github_handler.rb [hunk 4]
```diff
@@ -35,50 +39,59 @@ def handle
...
+      draft: pr_data.draft,
+      title: pr_data.title,
+      body: pr_data.body,
+      closed_at: pr_data.closed_at,
+      merged_at: pr_data.merged_at,
+      merged: pr_data.merged,
+      github_id: pr_data.id
+    )
+  rescue ActiveRecord::RecordNotUnique
+    Rails.logger.error('This pull request already exists!')
```
Comment (santib) [inline app/services/github_handler.rb]: As I mentioned in the previous comment I think it's more correct to just let the exception happen if we are not going to handle it. This log will even give us less information on what happened than the exception

Comment (hdamico) [reply app/services/github_handler.rb]: Ok!

**Pull request merged**